src/java.desktop/share/classes/javax/swing/TimerQueue.java

```diff
@@ -158,31 +158,27 @@ public void run() {
                     DelayedTimer runningTimer = queue.take();
                     Timer timer = runningTimer.getTimer();
                     timer.getLock().lock();
-                    try {
-                        DelayedTimer delayedTimer = timer.delayedTimer;
-                        if (delayedTimer == runningTimer) {
-                            /*
-                             * Timer is not removed (delayedTimer != null)
-                             * or not removed and added (runningTimer == delayedTimer)
-                             * after we get it from the queue and before the
-                             * lock on the timer is acquired
-                             */
-                            timer.post(); // have timer post an event
-                            timer.delayedTimer = null;
-                            if (timer.isRepeats()) {
-                                delayedTimer.setTime(now()
-                                    + TimeUnit.MILLISECONDS.toNanos(
-                                          timer.getDelay()));
-                                addTimer(delayedTimer);
-                            }
+                    DelayedTimer delayedTimer = timer.delayedTimer;
+                    if (delayedTimer == runningTimer) {
+                        /*
+                         * Timer is not removed (delayedTimer != null)
+                         * or not removed and added (runningTimer == delayedTimer)
+                         * after we get it from the queue and before the
+                         * lock on the timer is acquired
+                         */
+                        timer.post(); // have timer post an event
+                        timer.delayedTimer = null;
+                        if (timer.isRepeats()) {
+                            delayedTimer.setTime(now()
+                                + TimeUnit.MILLISECONDS.toNanos(
+                                      timer.getDelay()));
+                            addTimer(delayedTimer);
                         }
-
-                        // Allow run other threads on systems without kernel threads
-                        timer.getLock().newCondition().awaitNanos(1);
-                    } catch (SecurityException ignore) {
-                    } finally {
-                        timer.getLock().unlock();
                     }
+
+                    // Allow run other threads on systems without kernel threads
+                    timer.getLock().newCondition().awaitNanos(1);
+                    timer.getLock().unlock();
```

Comment:
It might be a good idea to keep the unlock in the finally block, even if there is no obvious candidate to throw an exception.


https://docs.oracle.com/en/java/javase/23/docs/api/java.base/java/util/concurrent/locks/Lock.html

> With this increased flexibility comes additional responsibility. The absence of block-structured locking removes the automatic release of locks that occurs with synchronized methods and statements. In most cases, the following idiom should be used:
>
>  <pre>
>  Lock l = ...;
>  l.lock();
>  try {
>    // access the resource protected by this lock
>  } finally {
>    l.unlock();
>  }
> </pre>